From: Jiri Pirko <jpirko@redhat.com>
Date: Tue, 23 Feb 2010 09:19:49 +0000
Subject: [PATCH 09/47] net: convert multiple drivers to use
 netdev_for_each_mc_addr, part5 V2

commit f9dcbcc9e338d08c0f7de7eba4eaafbbb7f81249 upstream.

removed some needless checks and also corrected bug in lp486e (dmi was passed
instead of dmi->dmi_addr)

Signed-off-by: Jiri Pirko <jpirko@redhat.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
[bwh: Restricted to drivers/net/r8169.c]
---
 drivers/net/r8169.c |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/drivers/net/r8169.c b/drivers/net/r8169.c
index d190827..616329f 100644
--- a/drivers/net/r8169.c
+++ b/drivers/net/r8169.c
@@ -4105,12 +4105,10 @@ static void rtl_set_rx_mode(struct net_device *dev)
 		mc_filter[1] = mc_filter[0] = 0xffffffff;
 	} else {
 		struct dev_mc_list *mclist;
-		unsigned int i;
 
 		rx_mode = AcceptBroadcast | AcceptMyPhys;
 		mc_filter[1] = mc_filter[0] = 0;
-		for (i = 0, mclist = dev->mc_list; mclist && i < netdev_mc_count(dev);
-		     i++, mclist = mclist->next) {
+		netdev_for_each_mc_addr(mclist, dev) {
 			int bit_nr = ether_crc(ETH_ALEN, mclist->dmi_addr) >> 26;
 			mc_filter[bit_nr >> 5] |= 1 << (bit_nr & 31);
 			rx_mode |= AcceptMulticast;
-- 
1.7.5.4

